src/pipert2/utils/publish_queue.py

```diff
@@ -0,0 +1,78 @@
+import os
+from queue import Full
+from functools import wraps
+from multiprocessing import Queue
+
+
+def ensure_parent(func):
+    @wraps(func)
+    def inner(self, *args, **kwargs):
+        if os.getpid() != self._creator_pid:
+            raise RuntimeError("{} can only be called in the "
+                               "parent.".format(func.__name__))
+        return func(self, *args, **kwargs)
+
+    return inner
+
+
+class PublishQueue(object):
+    """A class that wraps multiprocessing queue to make it possible to publish a single object to multiple queues.
+
+    """
+
+    def __init__(self):
+        self._queues = []
+        self._creator_pid = os.getpid()
+
+    @ensure_parent
+    def register(self):
+        """Register a new queue to publish to.
+
+        Returns:
+            A multiprocessing queue object.
+
+        """
+
+        q = Queue(maxsize=1)
+        self._queues.append(q)
+
+        return q
+
+    @ensure_parent
+    def put(self, value, block=False, timeout=1):
+        """Publish a value to every registered queue.
+
+        Args:
+            value: The value to push to every registered queue.
+            block: Whether or not to block the queue when putting a message.
+            timeout: How long to wait if block is true.
```

Comment:
Technically, the timeout will be the given time multiple by the output queues count.